### PR TITLE
feat: extending a docker image

### PR DESCRIPTION
**Pull request by FenrirUnbound**
# Additional topics

Added documentation on how you can modify the bookend plugins to change the order of the built-in plugins. It also refers to custom plugins that are written in a different section.

Added documentation on how to use the API Docker image as a base and build on top of it, such as including custom dependencies.

screen shot 2017-01-30 at 17 21 48

### CODE REVIEW

File: docs/cluster-management/configure-api.md [hunk 1]
```diff
@@ -49,6 +49,26 @@ auth:
...
+
+| Key | Default| Description |
+|:----|:-------|:------------|
+| BOOKENDS_SETUP | None | The ordered list of plugins to execute at the beginning of every build. Take the forms of `'["first", "second", ...]'` |
+| BOOKENDS_TEARDOWN | None | The ordered list of plugins to execute at the end of every build. Take the forms of `'["first", "second", ...]'` |
+
+
+```yaml
+# config/local.yaml
+ecosystem:
```
Comment (stjohnjohnson) [inline docs/cluster-management/configure-api.md]: I'm changing this to `bookends` here https://github.com/screwdriver-cd/screwdriver/pull/444

File: docs/cluster-management/configure-api.md [hunk 2]
```diff
@@ -49,6 +49,26 @@ auth:
         - batman
 ```
 
+### Bookend Plugins
+
+You can globally configure which built-in bookend plugins will be used during a build. By default, `sd-scm` is enabled to begin builds with a SCM checkout command.
```
Comment (stjohnjohnson) [inline docs/cluster-management/configure-api.md]: Switching to `scm` instead of `sd-scm`

**Review by stjohnjohnson — CHANGES_REQUESTED** [review]
Looks good otherwise!  Just a quick rename of those fields.

**Review by stjohnjohnson — APPROVED** [review]
I approve of my own changes as well :)

File: docs/cluster-management/configure-api.md [hunk 3]
```diff
@@ -49,6 +49,26 @@ auth:
         - batman
 ```
 
+### Bookend Plugins
+
+You can globally configure which built-in bookend plugins will be used during a build. By default, `scm` is enabled to begin builds with a SCM checkout command.
+
+If you're looking to include a custom bookend in the API, please refer to the *Using a custom bookend* topic in the *Extending the Docker container* section.
```
Comment (tkyi) [inline docs/cluster-management/configure-api.md]: Could link the section by doing 
```md
[Extending the Docker container](http://docs.screwdriver.cd/cluster-management/configure-api/#extending-the-docker-container)
```

File: docs/cluster-management/configure-api.md [hunk 4]
```diff
@@ -225,3 +245,57 @@ scm:
         oauthClientId: YOUR-APP-KEY
         oauthClientSecret: YOUR-APP-SECRET
 ```
+
+## Extending the Docker container
+
+There are some scenarios where you would prefer to extend the ScrewdriverCD Docker image, such as using custom Bookend plugins. This section is not meant to be exhaustive or complete, but will provide insight into some of the fundamental cases.
```
Comment (tkyi) [inline docs/cluster-management/configure-api.md]: `Screwdriver Docker image`*?

**Review by tkyi — CHANGES_REQUESTED** [review]

File: docs/cluster-management/configure-api.md [hunk 5]
```diff
@@ -225,3 +245,56 @@ scm:
         oauthClientId: YOUR-APP-KEY
         oauthClientSecret: YOUR-APP-SECRET
 ```
+
+## Extending the Docker container
+
+There are some scenarios where you would prefer to extend the Screwdriver.cd Docker image, such as using custom Bookend plugins. This section is not meant to be exhaustive or complete, but will provide insight into some of the fundamental cases.
+
+### Using a custom bookend
```
Comment (tkyi) [inline docs/cluster-management/configure-api.md:253]: Should the words be capitalized? It's okay. It looks okay.

**Review by tkyi — APPROVED** [review]
👍 Do the thing

**Review by d2lam — APPROVED** [review]
Looks good

**Pull request merged**